influxdb/resultset.py

```diff
@@ -91,14 +91,14 @@ def __len__(self):
     def keys(self):
         keys = []
         for serie in self._get_series():
-            keys.append((serie['name'], serie['tags']))
+            keys.append((serie.get('name', None), serie.get('tags', None)))
```

Comment:
"system" queries do not return a "name" , and "normal" queries (no "group by tag_name1(, tag_name2, ..)") doesn't return tags either.

NB: it's quite work in progress here ;)